lmctl/environment/lmenv.py

```diff
@@ -87,6 +93,23 @@ def _validate_oauth(cls, values):
             raise ValueError(f'Secure TNCO environment cannot be configured with "api_key" when using "auth_mode={OAUTH_MODE}". Use "client_id/client_secret" or "username/password" combination or set "auth_mode" to "{ZEN_AUTH_MODE}". If the TNCO environment is not secure then set "secure" to False')
         return values
 
+    @classmethod
+    def _validate_okta(cls, values):
+        client_id = values.get('client_id', None)
+        username = values.get('username', None)
+        if not client_id and not username:
+            raise ValueError(f'Secure TNCO environment must be configured with either "client_id" or "username" property when using "auth_mode={OKTA_MODE}". If the TNCO environment is not secure then set "secure" to False')
+        # Currently api_key can only be used with Zen, so we perform an extra check to let the user know
+        api_key = values.get('api_key', None)
+        if api_key is not None:
+            raise ValueError(f'Secure TNCO environment cannot be configured with "api_key" when using "auth_mode={OKTA_MODE}". Use "client_id/client_secret" or "username/password" combination or set "auth_mode" to "{ZEN_AUTH_MODE}". If the TNCO environment is not secure then set "secure" to False')
+        if not values.get('auth_server_id', None):
+            raise ValueError(f'Secure TNCO environment must be configured with "auth_server_id" when using "auth_mode={OKTA_MODE}". If the TNCO environment is not secure then set "secure" to False')
+        if values.get('username', None) and not values.get('scope', None):
+            raise ValueError(f'Secure TNCO environment cannot be configured with "scope" when using "auth_mode={OKTA_MODE}". If the TNCO environment is not secure then set "secure" to False')
```

Comment:
This says `Secure TNCO environment cannot be configured with "scope" when using "auth_mode={OKTA_MODE}"` but don't we actually want to say that scope must be set when using Okta mode with a username?

Reply:
Addressed.